apps/users-service/test/handlers/users.create-research-output.test.ts

```diff
@@ -0,0 +1,217 @@
+import Chance from 'chance';
+import nock from 'nock';
+import { handler } from '../../src/handlers/create-research-output';
+import { apiGatewayEvent } from '../helpers/events';
+import { APIGatewayProxyResult } from 'aws-lambda';
+import { createRandomUser } from '../helpers/create-user';
+import { config as authConfig } from '@asap-hub/auth';
+import { ResearchOutput } from '@asap-hub/model';
+
+jest.mock('@asap-hub/auth');
+
+const chance = new Chance();
+
+describe('GET /users/{id}/research-outputs', () => {
+  let id, code;
+
+  const output: ResearchOutput = {
+    url: chance.url(),
+    doi: chance.string(),
+    outputType: chance.pickone([
+      'dataset',
+      'code',
+      'protocol',
+      'resource',
+      'preprint',
+      'other',
+    ]),
+    title: chance.sentence(),
+    description: chance.paragraph(),
+    authors: [
+      {
+        name: chance.name(),
+      },
+    ],
+    publishDate: '2020-02-02T12:00:00Z',
+  };
+
+  beforeAll(async () => {
+    const user = await createRandomUser();
+    id = user.id;
+    code = user.connections[0].code;
+  });
+
+  test('return 401 when Authentication header is not set', async () => {
+    const result = (await handler(
+      apiGatewayEvent({
+        httpMethod: 'post',
+        body: output,
+        pathParameters: {
+          id,
```

Comment:
I'm wondering if we need the `id` in the header. We can extract it from the authenticated user.